### ISSUE TITLE
IsSteamRunningOnSteamDeck

### ISSUE THREAD

**Issue by 8bitskull** [britzl/steamworks-defold#27]
Probably blocked by same problem as https://github.com/britzl/steamworks-defold/issues/26, but would be nice to have this check:
https://partner.steamgames.com/doc/api/ISteamUtils#IsSteamRunningOnSteamDeck

**Comment by britzl** [comment]
Would you mind trying this branch with a Steam 1.55 update and see if it works well for you:

https://github.com/britzl/steamworks-defold/tree/dev-update-to-1.55

**Comment by 8bitskull** [comment]
I don't think it's working. I get this:

![image](https://user-images.githubusercontent.com/24439991/189648000-38a4035d-ce71-4f38-8782-daae3ab1e232.png)

Using this library:
https://github.com/britzl/steamworks-defold/archive/d9dee1c2c393bfe1c5dd841c3684249b70319d2f.zip

**Comment by britzl** [comment]
> Using this library:
> https://github.com/britzl/steamworks-defold/archive/d9dee1c2c393bfe1c5dd841c3684249b70319d2f.zip

Oh, that's not the right version. Please try the latest commit on the branch:

https://github.com/britzl/steamworks-defold/commits/dev-update-to-1.55

https://github.com/britzl/steamworks-defold/archive/bb4694af19f1ea15f33303bb1dac1136ea1eb161.zip

**Comment by 8bitskull** [comment]
Sorry about that!

OK, I created a Windows debug bundle and that works fine, outputs false for the steam deck check (as expected).

When trying to bundle release for Linux (i.e. to test on steam deck) I get this error log:
[log.txt](https://github.com/britzl/steamworks-defold/files/9548483/log.txt)

In the editor this seems to be the only error, the rest are warnings:
![image](https://user-images.githubusercontent.com/24439991/189668739-c18ab3fa-33ba-4dad-b8f8-6028f47ef732.png)

**Comment by britzl** [comment]
Ah, thanks. I've pushed a fix.

https://github.com/britzl/steamworks-defold/archive/145ef66a70a0c3d8bf3c105c95d315f63d3ea716.zip

**Comment by 8bitskull** [comment]
OK, it bundles now but crashes on launch on the Steam Deck. That's a release build, but I wouldn't know how to get debug output anyway. I do have the err extension in my project, but this particular crash doesn't generate any data (perhaps happens too quickly). 

A release build on Windows launches fine.

I have a workaround for now (system is linux + resolution matches Steam Deck + steamworks is active) so it's perhaps not an urgent fix. Happy to test more though if you have further ideas.

**Comment by britzl** [comment]
> I have a workaround for now (system is linux + resolution matches Steam Deck + steamworks is active) so it's perhaps not an urgent fix. Happy to test more though if you have further ideas.

I'm actually going to pick up a Steam Deck later today. I'll see if I can do some testing once I've familiarised myself with it. (What kind of developer options are there? Can I sideload apps while developing? etc)

**Comment by 8bitskull** [comment]
Ah, excellent! I have no idea unfortunately - I have simply been uploading test apps onto a beta branch of my Steam projects.

**Comment by 8bitskull** [comment]
Getting (rare, thankfully) comments that indicate my workaround sometimes fails - which I imagine either means the user is running an unusual operating system or has somehow changed the resolution on the Steam Deck. Accordingly, that means the issue is more relevant for me again, as having access to IsSteamRunningOnSteamDeck would be much more reliable.

**Comment by whiteboxdev** [comment]
+1 for this issue, running into the same problem.
